src/coffee/mappings.coffee

```diff
@@ -13,14 +13,36 @@ class CustomFieldMappings
     _.each fieldDefinitions, (fieldDefinition) =>
       if fieldDefinition.name is key
         switch fieldDefinition.type.name
-          when 'Number' then result = @mapNumber value,typeDefinitionKey,rowIndex
-          when 'Boolean' then result = @mapBoolean value,typeDefinitionKey,rowIndex
-          when 'Money' then result = @mapMoney value,typeDefinitionKey,rowIndex
-          when 'LocalizedString' then result = @mapLocalizedString value, typeDefinitionKey, rowIndex,langHeader
-          when 'Set' then result = @mapSet value,typeDefinitionKey,rowIndex,fieldDefinition.type.elementType
-          else result = value
+          when 'Number' then result = @mapNumber value, typeDefinitionKey, rowIndex
+          when 'Boolean' then result = @mapBoolean value, typeDefinitionKey, rowIndex
+          when 'Money' then result = @mapMoney value, typeDefinitionKey, rowIndex
+          when 'LocalizedString' then result = @mapLocalizedString value, typeDefinitionKey, rowIndex, langHeader
+          when 'Set' then result = @mapSet value, typeDefinitionKey, rowIndex
+          when 'String', 'Enum', 'LocalizedEnum', 'Date', 'Time', 'DateTime', 'Reference'
+            if (!_.isUndefined(value))
+              result = value
+          else
+            @errors.push "[row #{rowIndex}:#{typeDefinitionKey}] The type '#{fieldDefinition.type.name}' is not valid."
```

Comment:
You should add a `return` as the final statement in the `else`, else the error is pushed to the result, and we do not want that to happen.

Also, update the error message. The error here doesn't necessary mean that the type is invalid, `not supported` looks more appropriate